### PR TITLE
New Cheatsheet: radare2

### PR DESCRIPTION
**Pull request by therealkbhat**
**What does your Instant Answer do?**
Serves as a quick reference for radare2 users and developers. radare2 (or r2) is a reverse engineering framework and binary analysis tool. 

**What is the data source for your Instant Answer? (Provide a link if possible)**
<https://github.com/radare/radare2book/blob/master/refcard/radare2_rc.pdf>

**Why did you choose this data source?**
It was released today and covers all major commands. Interest in this cheatsheet was expressed [here](https://twitter.com/therealkbhat/status/686547901628039168).

**Are there any other alternative (better) data sources?**
Not that I know of.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
r2 users, developers and CTF (capture the flag) players around the world.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No... does [this](https://duck.co/ia/view/radare2_cheatsheet) count?

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

screenshot from 2016-01-11 22 59 51
screenshot from 2016-01-11 23 00 08
screenshot from 2016-01-11 23 00 35

Link to IA page: [here](https://duck.co/ia/view/radare2_cheatsheet).

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by lxndio** [comment]
The page you named at "Instant Answer idea" does not count, it is the IA page, so it is not connected to an IA idea.
Everything else seems to be good except for the name field which should be just "Radare2" instead of "radare2 Cheat Sheet".

**Comment by therealkbhat** [comment]
@lxndlo I made the necessary change.

Thanks for the info! A lot of the page layout has changed since I last contributed.

**Comment by moollaza** [comment]
@lxndio thanks for assisting! 

@therealkbhat hi again, and thanks for the contribution -- did you change your username?

File: share/goodie/cheat_sheets/json/radare2.json [hunk 1]
```diff
@@ -0,0 +1,359 @@
+{
+    "id": "radare2_cheat_sheet",
+    "name": "Radare2",
+    "description": "Commands for radare2, a framework for reverse engineering and binary analysis",
+    "metadata": {
+        "sourceName": "Github",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/radare2.json]: nit: should be `GitHub`

**Comment by therealkbhat** [comment]
@moollaza fixed; and yes, I did change my username (from kbhat95).

File: share/goodie/cheat_sheets/json/radare2.json [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+    "id": "radare2_cheat_sheet",
+    "name": "Radare2",
+    "description": "Commands for radare2, a framework for reverse engineering and binary analysis",
+    "metadata": {
+        "sourceName": "GitHub",
+        "sourceUrl": "https://github.com/radare/radare2book/blob/master/refcard/radare2_rc.pdf"
+    },
+    "aliases": ["r2"],
+    "template_type": "code",
+    "section_order": ["Survival Guide", "Flagspaces", "Flags", "Info", "Print string", "Visual mode", "Searching", "Saving", "Usable variables in expression"],
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/radare2.json]: @therealkbhat I think we can capitalise s in "Print string" and capitalise m in "Visual mode"

File: share/goodie/cheat_sheets/json/radare2.json [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+        }, {
+            "val": "show flag length",
+            "key": "fl"
+        }, {
+            "val": "show hexdump of flag",
+            "key": "fx"
+        }, {
+            "val": "set flag comment",
+            "key": "fC <name> <cmt>"
+        }],
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/radare2.json]: @therealkbhat s Lets start all val fields with a capital letter.

File: share/goodie/cheat_sheets/json/radare2.json [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+            "key": "[Enter]"
+        }, {
+            "val": "step / step over",
+            "key": "sS"
+        }, {
+            "val": "follow address of jump/call",
+            "key": "[Enter]"
+        }, {
+            "val": "go/seek to given offset",
+            "key": "o"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/radare2.json]: @therealkbhat same here .start val fields with a capital letter

File: share/goodie/cheat_sheets/json/radare2.json [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+            "key": ">||<"
+        }, {
+            "val": "(i)nsert code",
+            "key": "i"
+        }, {
+            "val": "(a)ssemble code",
+            "key": "a"
+        }, {
+            "val": "visual (A)ssembler",
+            "key": "A"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/radare2.json]: @therealkbhat  is enclosing the letter in brackets important?
If not we can avoid it

File: share/goodie/cheat_sheets/json/radare2.json [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+            "key": "ie"
+        }, {
+            "val": "Display sections",
+            "key": "iS"
+        }, {
+            "val": "Display relocations",
+            "key": "ir"
+        }],
+        "Print string": [{
+            "val": "Print stringZ'",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/radare2.json]: @therealkbhat  Is the z at the end a typo

**Comment by jophab** [comment]
@therealkbhat Can we remove some content to make it a little small

**Comment by therealkbhat** [comment]
@jophab, why is the size a problem currently? Looks fine to me.

**Comment by therealkbhat** [comment]
@moollaza what do you think?

**Comment by jophab** [comment]
@therealkbhat Size becomes concern for cheat sheets having LOC more than 300.
This cheat sheet is not so large. But still a little concering.
If you can avoid unimportant contents we can make it more compact.
I think it will make users pleasant if they see only important things

**Comment by therealkbhat** [comment]
@jophab fair enough, but I still feel that this cheatsheet is fine. All the commands are relevant, and the whole point of a cheatsheet is to have these commands for ready reference.

Perhaps we can get a third opinion on this?

**Comment by zekiel** [comment]
I think there was a valuable conversation here --- what should the quality check on content length be? The basic, "instruction" for developers is to include the most essential/useful information in cheat sheets but that can sometimes be a lot. If there could be a basic rule (e.g. between 200 and 400 lines or 10 lines per section) -- just spitballing :) 

Thanks for the contribution @therealkbhat

**Comment by therealkbhat** [comment]
@zekial makes sense. :)

What do you think about this sheet in particular?

**Comment by jophab** [comment]
@zekiel
I think what you mean is to keep all the contents.
@therealkbhat Lets keep them :+1: 
Sorry about unnecessary  suggestions

**Comment by zekiel** [comment]
@jophab  @therealkbhat  lgtm! :D

**Comment by jophab** [comment]
@zekiel @therealkbhat :+1:  LGTM :smiley:

**Comment by moollaza** [comment]
@therealkbhat @jophab Thanks! This LGTM :+1:

I'm not sure why the test is failing for Beta, I tested manually and it seems fine.

**Pull request merged**